tests/ert_tests/ert3/stats/test_distributions.py

```diff
@@ -237,3 +274,56 @@ def test_uniform_ppf(lower, upper, q, size, index):
     assert sorted(dist.index) == sorted(ppf_result.index)
     for idx in dist.index:
         assert ppf_result.data[idx] == pytest.approx(expected_value)
+
+
+@pytest.mark.parametrize(
+    ("valueset", "q", "expected_val"),
+    (
+        ([0], 0.0, np.nan),
+        ([0], 0.1, 0),
+        ([0], 0.2, 0),
+        ([0], 0.5, 0),
+        ([1], 0.9, 1),
+        ([42], 0.999, 42),
+        ([0], 1.1, np.nan),
+        ([0], 0.1, 0),
+        ([42, 420, 4200], 0.0, np.nan),
+        ([42, 420, 4200], 0.2, 42),
+        ([42, 420, 4200], 0.332, 42),
+        ([42, 420, 4200], 0.334, 420),
+        ([42, 420, 4200], 0.5, 420),
+        ([42, 420, 4200], 0.665, 420),
+        ([42, 420, 4200], 0.667, 4200),
+        ([42, 420, 4200], 0.9, 4200),
+        ([42, 420, 4200], 0.9, 4200),
+        ([42, 420, 4200], 1.1, np.nan),
+    ),
+)
+def test_discrete_ppf(valueset, q, expected_val):
+
+    for size in (1, 5, 10):
+        dist = ert3.stats.Discrete(valueset, size=size)
+        expected = [expected_val] * size
+        assert len(dist.index) == len(expected)
```

Comment:
Given the test on the line 310, this one seems a bit redudant.

Reply:
Hmm, I would say 311 and 312 make 307 and 310 redundant, but when bugs pop up, it might be easier to have them triggered early in simpler tests, so I suggest to keep.